tardis/plasma/properties/ion_population.py

```diff
@@ -136,36 +137,31 @@ class RadiationFieldCorrection():
     This function adds a field 'delta' to the phi table given to the function
 
     """
-
-    def calculate(self, w, departure_coefficient=None, chi_0_species=(20, 2)):
-        #factor delta ML 1993
-        if departure_coefficient is None:
-            departure_coefficient = 1. / self.ws
-
-        ionization_data = self.atom_data.ionization_data
-
+    @staticmethod
+    def calculate(w, ionization_data, beta_rad, t_electron, t_rad,
+        beta_electron, levels):
+        # factor delta ML 1993
+        departure_coefficient = 1. / w
+        chi_0_species=(20, 2)
         chi_0 = ionization_data.ionization_energy.ix[chi_0_species]
-        radiation_field_correction = -np.ones((len(ionization_data), len(self.beta_rads)))
+        radiation_field_correction = -np.ones((len(ionization_data), len(
+            beta_rad)))
         less_than_chi_0 = (ionization_data.ionization_energy < chi_0).values
-
-        factor_a =  (self.t_electrons / (departure_coefficient * self.ws * self.t_rads))
-
+        factor_a = (t_electron / (departure_coefficient * w * t_rad))
         radiation_field_correction[~less_than_chi_0] = factor_a * \
-                                     np.exp(np.outer(ionization_data.ionization_energy.values[~less_than_chi_0],
-                                                     self.beta_rads - self.beta_electrons))
-
-
-
-
-        radiation_field_correction[less_than_chi_0] = 1 - np.exp(np.outer(ionization_data.ionization_energy.values
-                                                                      [less_than_chi_0], self.beta_rads)
-                                                                 - self.beta_rads * chi_0)
+            np.exp(np.outer(ionization_data.ionization_energy.values[
+            ~less_than_chi_0], beta_rad - beta_electron))
+        radiation_field_correction[less_than_chi_0] = 1 - np.exp(np.outer(
+            ionization_data.ionization_energy.values[less_than_chi_0],
+            beta_rad) - beta_rad * chi_0)
         radiation_field_correction[less_than_chi_0] += factor_a * np.exp(
-            np.outer(ionization_data.ionization_energy.values[less_than_chi_0], self.beta_rads) -
-             chi_0*self.beta_electrons)
-
-        return pd.DataFrame(radiation_field_correction, columns=np.arange(len(self.t_rads)),
+            np.outer(ionization_data.ionization_energy.values[less_than_chi_0],
+                     beta_rad) - chi_0 * beta_electron)
+        delta = pd.DataFrame(radiation_field_correction,
+                            columns=np.arange(len(t_rad)),
                             index=ionization_data.index)
+        delta.drop([(1,1), (4,3)], inplace=True)
```

Comment:
Something weird going on with indexing here. The `delta` dataframe is generated with two extra ions compared to the `phi` ones. I understand the (1,1) one because the ionization data isn't being filtered (something to change), but I don't understand where the (4,3) comes from.